aptos-move/move-examples/nft_auction_house/sources/auction_house_own_listing.move

```diff
@@ -0,0 +1,531 @@
+    const EBID_INCREASE_TOO_SMALL: u64 = 4;
+
+    /// Auction ended
+    const EAUCTION_ENDED: u64 = 5;
+
+    /// Minimal incremental should be bigger than 0
+    const EBID_MIN_INCREMENTAL_IS_ZERO: u64 = 6;
+
+    /// Bid not found
+    const EBID_NOT_FOUND_FOR_AUCTION: u64 = 7;
+
+    /// Reserved operation for auction house owner
+    const EONLY_AUCTION_HOUSE_OWNER_CAN_PERFORM_THIS_OPERATION: u64 = 8;
+
+    /// Auction not ended
+    const EAUCTION_NOT_ENDED: u64 = 9;
+
+    struct AuctionHouseConfig has key {
+        market_fee_numerator: u64,
+        market_fee_denominator: u64,
+        fee_address: address,
+    }
+
+    struct Auctions<phantom CoinType> has key {
+        cur_auction_id: u64, // this is used to generate next auction_id
+        all_active_auctions: Table<u64, Auction<CoinType>>,
+        listing_event: EventHandle<ListingEvent>,
+        bid_event: EventHandle<BidEvent>,
+        cancel_bid_events: EventHandle<CancelBidEvent>
+    }
+
+    struct BidEvent has copy, drop, store {
+        market_address: address,
+        bid_id: ID,
+    }
+
+    struct CancelBidEvent has copy, drop, store {
+        market_address: address,
+        bid_id: ID,
+    }
+
+    struct Auction<phantom CoinType> has drop, store {
+        listing: Listing<CoinType>,
+        bids: vector<ID>,
+        highest_price: u64,
+        highest_bid: ID,
+        min_incremental_price: u64,
+    }
+
+    public entry fun initialize_auction_house(
+        account: &signer,
+        market_fee_numerator: u64,
+        market_fee_denominator: u64,
+        fee_address: address,
+    ) {
+        move_to(
+            account,
+            AuctionHouseConfig {
+                market_fee_denominator,
+                market_fee_numerator,
+                fee_address,
+            }
+        );
+    }
+
+    public entry fun initialize_auction<CoinType>(account: &signer) {
+        move_to(
+            account,
+            Auctions<CoinType> {
+                cur_auction_id: 0,
+                all_active_auctions: table::new(),
+                listing_event: account::new_event_handle<ListingEvent>(account),
+                bid_event: account::new_event_handle<BidEvent>(account),
+                cancel_bid_events: account::new_event_handle<CancelBidEvent>(account),
+
+            }
+        );
+    }
+
+    public fun generate_auction_data<CoinType>(
+        owner: &signer,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        start_sec: u64, // specify when the auction starts
+        expiration_sec: u64, // specify when the auction ends
+        withdraw_expiration_sec: u64,
+        min_incremental_price: u64,
+    ): Auction<CoinType> {
+        let sec = timestamp::now_seconds();
+        assert!(sec <= start_sec, error::invalid_argument(EINVALID_START_TIME));
+        assert!(start_sec < expiration_sec, error::invalid_argument(EINVALID_EXPIRATION_TIME));
```

Comment:
What about withdraw_expiration_sec?

Reply:
that is checked in `create_list` to ensure withdraw_expiration_sec larger than expiration_sec